core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -151,12 +152,16 @@ public void process(final Archive<?> applicationArchive, final TestClass testCla
         tryToAdd(libs, libraryContainer, "WEB-INF/lib/scala-library-", "cucumber.api.scala.ScalaDsl", "scala.App");
     }
 
-    private static void addConfiguration(final JavaArchive resourceJar, final CucumberConfiguration cucumberConfiguration, final boolean report, final String reportDirectory) {
+    private static void addConfiguration(final JavaArchive resourceJar, final CucumberConfiguration cucumberConfiguration, final boolean report, final String reportDirectory, final String objectFactory) {
         final StringBuilder config = new StringBuilder();
         config.append(CucumberConfiguration.COLORS).append("=").append(cucumberConfiguration.isColorized()).append("\n")
                 .append(CucumberConfiguration.REPORTABLE).append("=").append(report).append("\n")
                 .append(CucumberConfiguration.REPORTABLE_PATH).append("=")
                 .append(reportDirectory == null ? null : new File(reportDirectory).getAbsolutePath()).append("\n");
+
+        if (objectFactory != null) {
+             config.append(CucumberConfiguration.OBJECT_FACTORY).append("=").append(objectFactory).append("\n");

```

Comment:
Can you use the getter to remove the new param please? Otherwise looks very good!